src/OpenTelemetry/OpenTelemetrySdk.cs

```diff
@@ -0,0 +1,145 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+#if EXPOSE_EXPERIMENTAL_FEATURES && NET8_0_OR_GREATER
+using System.Diagnostics.CodeAnalysis;
+#endif
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Internal;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry;
+
+/// <summary>
+/// Contains methods for configuring the OpenTelemetry SDK and accessing
+/// logging, metrics, and tracing providers.
+/// </summary>
+public sealed class OpenTelemetrySdk : IDisposable
+{
+    private readonly ServiceProvider serviceProvider;
+
+    private OpenTelemetrySdk(
+        Action<IOpenTelemetryBuilder> configure)
+    {
+        Debug.Assert(configure != null, "configure was null");
+
+        var services = new ServiceCollection();
+
+        var builder = new OpenTelemetrySdkBuilder(services);
+
+        configure!(builder);
+
+        this.serviceProvider = services.BuildServiceProvider();
+
+        this.LoggerProvider = (LoggerProvider?)this.serviceProvider.GetService(typeof(LoggerProvider))
+            ?? new NoopLoggerProvider();
+        this.MeterProvider = (MeterProvider?)this.serviceProvider.GetService(typeof(MeterProvider))
+            ?? new NoopMeterProvider();
+        this.TracerProvider = (TracerProvider?)this.serviceProvider.GetService(typeof(TracerProvider))
+            ?? new NoopTracerProvider();
+    }
+
+    /// <summary>
+    /// Gets the <see cref="IServiceProvider"/> containing SDK services.
+    /// </summary>
+    public IServiceProvider Services => this.serviceProvider;
```

Comment:
Here is the `GetLoggerFactory` extension: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5325/files#diff-3b6f110a0b1b70153d7ebf99eee6fe13464cc3b0172013f50fdd7ba4e144e137R33

The `IServiceProvider` is really _the_ extension point for `OpenTelemetrySdk`. If someone wants to add something to `IOpenTelemetryBuilder.Services` (`IServiceCollection`) during build and then let users read it back/access it, this is the way to do it.

Not sure if it will really be needed. I went ahead and made it `internal` for now. We can always bring it back if needed.